README.md

```diff
@@ -1,23 +1,147 @@
-``` sh
-{minizinc -c -G ../../../share/minizinc/flatzingo --output-fzn-to-stdout example.mzn | fzn2lp; cat static_check.lp types.lp }| clingcon
+
+## Install
+
+### Using Docker
+
+Follow the steps in the last [section](install-using-a-docker-image). *Note that **Usage** will vary on the paths.*
+
+### Manual Installation
+
+1. Install MiniZinc as showed [here](https://www.minizinc.org/doc-2.5.5/en/installation.html)
+2. MiniZinc requires the [files](share/minizinc/flatzingo) defining the high-level constraints used by flatzingo, to be on a relative path to the installation of MiniZinc. So they have to be copied to these directory. Given the MiniZinc path added during the instalation `$PATH_MINIZINC` copy the files with: 
+```
+cp share/minizinc/flatzingo/* $PATH_MINIZIC/share/minizinc/flatzingo
+```
+Example in MacOS
+```
+cp share/minizinc/flatzingo/* /Applications/MiniZincIDE.app/Contents/Resources/share/minizinc/flatzingo
+```
+3. [Install fzn2lp](https://github.com/potassco/fzn2lp) manually using `cargo`
+4. [Install clingcon](https://github.com/potassco/clingcon) 
+
+## Usage
+
+Run MiniZinc (Without solving `-c`) to get a FlatZinc output (`--output-fzn-to-stdout`) from a MiniZinc model (in this case `encodings/example.mzn`). Additionally the path to the high-level constraints (moved during the installation) must be specified with the `-G` option.
+The output is then piped to `fzn2lp`
+
+```
+minizinc  -G -c flatzingo --output-fzn-to-stdout encodings/example.mzn | fzn2lp > tmp.lp
+```
+
+Use the `lp` file obtained in a `clingcon` call with other encodings. 
+
+- **Static check**
+  
+Check the problem before converting it to clingcon's syntax
+```
+clingcon static_check.lp types.lp tmp.lp
 ```
 
-computing a solution,
-``` sh
-{minizinc -c -G ../../../share/minizinc/flatzingo --output-fzn-to-stdout example.mzn | fzn2lp; cat encoding.lp types.lp }| clingcon
+- **Compute solution with `clingcon`**
+
+```
+clingcon encodings/encoding.lp encodings/types.lp tmp.lp
+```
+
+### Single command
+
+The full process can be pied a single command:
 ```
+{minizinc -c -G flatzingo --output-fzn-to-stdout encodings/example.mzn | fzn2lp; cat encodings/encoding.lp encodings/types.lp }| clingcon
+```
+
+## Set up for MiniZinc competition with docker
+
+### Install using a docker image
+
+The docker image is defined in [Dockerfile_Flatzingo](Dockerfile_Flatzingo). It contains the necessary settings to build up the environment for minizinc using flatzingo.
+This option builds up an ubuntu OS where it installs the specific solver requirements `fzn2lp` and `flatzingo`. In a next stage it builds up on the docker image used of MiniZinc competition and moves the flatzingo files to be relative to the MiniZic installation. 
 
-Create a docker image:
+#### Create a docker image
 
+We create a docker image (without cache, so that every new build builds everything from scratch) with the name `flatzingo` and tag `1.0` based on the DockerFile 
+
+```
```

Comment:
I'm not familiar with the tagging. Most recent tag is https://hub.docker.com/r/maxostrowski/flatzingo 1.6. I do not know what to take when doing this only on a local machine.

Reply:
I undestood that in this case the tag is independent and just used for the name of the image you create.